src/attr/_make.py

```diff
@@ -597,6 +599,48 @@ def _transform_attrs(
     return _Attributes((AttrsClass(attrs), base_attrs, base_attr_map))
 
 
+def _make_cached_property_getattr(
+    cached_properties,
+    cls,
+):
+    lines = [
+        # Wrapped to get `__class__` into closure cell for super()
+        # (It will be replaced with the newly constructed class after construction).
+        "def wrapper(_cls, cached_properties, _cached_setattr_get):",
+        "    __class__ = _cls",
+        "    def __getattr__(self, item):",
```

Comment:
You can make `cached_properties` be a local instead of a global inside of `__getattr__` by making it an optional parameter with `cached_properties` as the default value.

It's an old trick for speeding up variable access.

Reply:
I'm presuming this applies to the other non-local references inside `__getattr__`, so I've added them all.  I've also removed them from `wrapper`, as that seems unnecessary if they're then being localised.